sessions/recap-project-6-NEW/activity-planner/user-stories/2_activity-details.md

```diff
@@ -23,13 +23,44 @@ Group Todo: Add wireframes
   - Activity categories
   - Activity area
   - Activity country
-  - Creation date
-  - Last update date
 - The detailed view includes a "back" link, allowing users to easily navigate back to the list of activities.
 - The page implements proper error handling for non-existent activity IDs.
 - The page shows a loading state while fetching activity details.
 
 ## Tasks
 
 - [ ] Create feature branch `feature/activity-details`
-- [ ] Group Todo: Add tasks
\ No newline at end of file
+- [ ] Create API endpoint for fetching single activity by ID
+  - [ ] Implement MongoDB findById query
+  - [ ] Add proper error handling for invalid IDs
+  - [ ] Add validation for required fields
+- [ ] Create frontend components
+  - [ ] Create `ActivityDetails` page component
+  - [ ] Create `ActivityInfo` component for displaying activity data
+  - [ ] Create `BackButton` component for navigation
+  - [ ] Design props interface for `ActivityInfo`:
+    - `activity`: object with fields (title, imageUrl, description, categories, area, country)
+- [ ] Implement dynamic routing and navigation
+  - [ ] Set up dynamic route for `/activities/[id]`
+  - [ ] Extract activity ID from URL parameters using `useRouter`
+  - [ ] Implement navigation from activity list to details page
+  - [ ] Add back navigation functionality
+- [ ] Implement data fetching and state management
+  - [ ] Use SWR hook to fetch activity data: `useSWR('/api/activities/[id]', fetcher)`
+  - [ ] Handle SWR loading state (`isLoading`) and error state (`error`)
+  - [ ] Implement conditional rendering based on SWR states
+  - [ ] Handle SWR data validation and transformation
+  - [ ] Implement URL parameter validation
+- [ ] Add loading and error states
+  - [ ] Create loading state component/UI for data fetching
+  - [ ] Create error state component/UI for invalid IDs or network errors
+  - [ ] Create "not found" state for non-existent activities
+  - [ ] Implement conditional rendering based on state
+- [ ] Implement date formatting
+  - [ ] Format creation and update dates for display
+  - [ ] Use appropriate date formatting library or native methods
+- [ ] Style components
+  - [ ] Design detailed view layout with proper typography
+  - [ ] Style activity information display with clear visual hierarchy
+  - [ ] Implement responsive design for mobile and desktop
+  - [ ] Style navigation elements (back button)
```

Comment:
we explicitly tell the students to only create a mobile view first. this does not fit this constraint. 
"Design detailed view layout with proper typgraphy"? This is way to detailed imo.